### PR TITLE
SUMO-103799: refactor RealisticSearchClient

### PR DESCRIPTION
**Pull request by tgimis**
Refactor RealisticSearchClient

### CODE REVIEW

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala [hunk 1]
```diff
@@ -144,36 +48,7 @@ class RestlasticSearchClient2(endpointProvider: EndpointProvider, signer: Option
...
-  }
-
-  def deleteDocuments(index: Index, tpe: Type, deleteQuery: QueryRoot, pluginEnabled: Boolean = false): Future[Map[Index, DeleteResponse]] = {
-    def firstScroll(scId: ScrollId) = startScrollRequest(index, tpe, deleteQuery)
-
-    scrollDelete(index, tpe, ScrollId(""), Map.empty[Index, DeleteResponse], firstScroll)
-  }
-
-  private def scrollDelete(index: Index,
+  override protected def scrollDelete(index: Index,
```
Comment (michaljmatusiak92) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala:51]: Please fix also the indentations of the arguments

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala [hunk 2]
```diff
@@ -202,44 +77,21 @@ class RestlasticSearchClient2(endpointProvider: EndpointProvider, signer: Option
   // Scroll requests have optimizations that make them faster when the sort order is _doc.
   // Put sort by _doc in query as described in the the following document
   // https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
-  def startScrollRequest(index: Index,
+  override def startScrollRequest(index: Index,
```
Comment (michaljmatusiak92) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala:89]: as well here

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala [hunk 3]
```diff
@@ -202,44 +77,21 @@ class RestlasticSearchClient2(endpointProvider: EndpointProvider, signer: Option
...
+    super.startScrollRequest(index, tpe, query, resultWindowOpt, fromOpt, sizeOpt, preference, params)
   }
 
-  def refresh(index: Index): Future[RawJsonResponse] = {
-    implicit val ec = indexExecutionCtx
-    runEsCommand(EmptyObject, s"/${index.name}/_refresh")
-  }
-
-  def runRawEsRequest(op: String,
+  override def runRawEsRequest(op: String,
```
Comment (michaljmatusiak92) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala:103]: and in the rest of the methods

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala (hunk 1 above)
Comment (tgimis) [reply elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala:51]: Fixed

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala (hunk 2 above)
Comment (tgimis) [reply elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala:89]: Fixed

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala (hunk 3 above)
Comment (tgimis) [reply elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient2.scala:103]: Fixed

**Pull request merged**